src/bhv_basic_move.cpp

```diff
@@ -105,13 +107,21 @@ Bhv_BasicMove::execute( PlayerAgent * agent )
     agent->debugClient().setTarget( target_point );
     agent->debugClient().addCircle( target_point, dist_thr );
 
-    if ( ! Body_GoToPoint( target_point, dist_thr, dash_power
-                           ).execute( agent ) )
-    {
-        Body_TurnToBall().execute( agent );
+
+    if ( Body_GoToPoint( target_point, dist_thr, dash_power
+			 ).execute( agent ) ||
+	 Body_TurnToBall().execute( agent ) ) {
+      if (ball.rposValid() || wm.self().collidesWithPost()) {
+	success = true;
+      } else {
+	success = false;
+      }
+    } else {
+      success = false;
     }
 
-    if ( wm.existKickableOpponent()
+    if ( wm.existKickableOpponent() &&
+	 ball.rposValid()
```

Comment:
Why are we shortcutting these actions? Both actions already account for the fact that the ball position may not be valid.